journals/apps/journals/models.py

```diff
@@ -309,6 +309,22 @@ class JournalDocument(AbstractDocument):
 
     admin_form_fields = Document.admin_form_fields
 
+    @property
+    def deletion_warning_message(self):
+        """
+        returns the custom deletion warning message if document is being used in journal pages.
+        """
+        warning_message = "Are you sure you want to delete this document?"
+        journal_pages_title = JournalPage.objects.filter(documents=self).values_list('title', flat=True)
+        if journal_pages_title:
+            warning_message = "The document '{document_title}' is being used in page(s): {page_titles} " \
```

Comment:
use `pluralize` or equivalent for strings, see https://dustindavis.me/django-pluralize-outside-templates/